metadeploy/api/models.py

```diff
@@ -847,6 +862,56 @@ def run(self, ctx, plan, steps, org):
         flow_coordinator.run(org)
 
 
+class ScratchOrg(HashIdMixin, models.Model):
+    Status = Choices("started", "complete", "failed", "canceled")
+
+    plan = models.ForeignKey(Plan, on_delete=models.CASCADE)
+    email = models.EmailField()
+
+    enqueued_at = models.DateTimeField(null=True)
+    job_id = models.UUIDField(null=True)
+    # This is set in a user's session to let them continue to access
+    # this job, without being otherwise auth'd:
+    uuid = models.UUIDField(null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    edited_at = models.DateTimeField(auto_now=True)
+    status = models.CharField(choices=Status, max_length=64, default=Status.started)
+    config = JSONField(null=True, blank=True, encoder=DjangoJSONEncoder)
+    org_id = models.CharField(null=True, blank=True, max_length=18)
+
+    def save(self, *args, **kwargs):
+        ret = super().save(*args, **kwargs)
+        if not self.enqueued_at:
+            from .jobs import create_scratch_org_job
+
+            job = create_scratch_org_job.delay(
+                plan_id=str(self.plan.id),
+                email=self.email,
+                org_name=self.plan.org_name,
+                result_id=self.id,
+            )
+            self.job_id = job.id
+            self.enqueued_at = job.enqueued_at
+            # Yes, this bounces two saves:
+            super().save()
+        return ret
+
+    def subscribable_by(self, user):  # pragma: nocover
+        return True
```

Comment:
@wlonk Hm. I wonder if, similar to `scratch_org_get`, we want to instead check in `consumers.py` if the subscribing session has the uuid for this org?